compiler/noirc_frontend/src/hir/comptime/value.rs

```diff
@@ -609,3 +637,204 @@ fn display_trait_constraint(interner: &NodeInterner, trait_constraint: &TraitCon
     let trait_ = interner.get_trait(trait_constraint.trait_id);
     format!("{}: {}{}", trait_constraint.typ, trait_.name, trait_constraint.trait_generics)
 }
+
+// Returns a new Expression where all Interned and Resolved expressions have been turned into non-interned ExpressionKind.
+fn resolve_expression(interner: &NodeInterner, expr: Expression) -> Expression {
+    Expression { kind: resolve_expression_kind(interner, expr.kind), span: expr.span }
+}
+
+// Returns a new ExpressionKind where all Interned and Resolved expressions have been turned into non-interned ExpressionKind.
+fn resolve_expression_kind(interner: &NodeInterner, expr: ExpressionKind) -> ExpressionKind {
```

Comment:
These are only used for display, right? If so, can we use `to_display_ast` instead? Or does that not cover some cases?

Reply:
The scenario this is handling is if you have an interned expression and you put it inside something else, for example an array literal:

```rust
let expr = quote { 1 }.as_expr().unwrap();
let array = quote { [ $expr, $expr ] }.as_expr().unwrap();
println(array);
```

Before doing all of that `resolve_expression_kind` that printed this:

```
[?Interned, ?Interned, ?Interned]
```

Even if I checked if the expression was an interned and resolved that, it would print that. At first I was confused, but the thing is that the expression was an `ArrayLiteral` that had interned expressions inside it. So I had two options:
1. Implement `fmt` again for all of the expression kinds, turning any inner interned ones into non-interned ones.
2. Only convert an `ExpressoinKind` into another one by mapping interned sub-expressions into non-interned ones,  then calling `fmt` on that.

I chose 2 to avoid repeating the printing logic.

Reply:
(we could also just leave it like `[?Interned, ?Interned, ?Interned]` but while coding the "inject" example, and the "mutate" method, I used print-debugging a lot and it was very hard to understand what was going on without unwrapping these expressions.

Reply:
Ah, ok. I misunderstood the method and thought we could replace it entirely with to_display_ast because of the Interned case.

Can we rename this method to remove the `resolve_` suffix since that is used by name resolution? We can call it something alluding to removing interned expressions, following interned expressions or similar.